metaflow/flowspec.py

```diff
@@ -43,13 +43,19 @@ class FlowSpec(object):
     # Name starting with '__', methods, functions and Parameters do not need
     # to be listed.
     _EPHEMERAL = {'_EPHEMERAL',
+                  '_NON_PARAMETERS',
                   '_datastore',
                   '_cached_input',
                   '_graph',
                   '_flow_decorators',
                   '_steps',
                   'index',
                   'input'}
+    # When checking for parameters, we look at dir(self) but we want to exclude
+    # attributes that are definitely not parameters and may be expensive to
+    # compute (like anything related to the `foreach_stack`). We don't need to exclude
+    # names starting with `_` as those are already excluded from `_get_parameters`.
+    _NON_PARAMETERS = {'cmd', 'foreach_stack', 'index', 'input', 'script_name', 'name'}
```

Comment:
A note for a follow up PR - we should throw an error if we detect that a user is using any of these as parameter names.